patroni/api.py

```diff
@@ -817,13 +1352,29 @@ def __members_ips(self):
                         except Exception as e:
                             logger.debug('Failed to parse url %s: %r', member.api_url, e)
 
-    def check_access(self, rh):
+    def check_access(self, rh: RestApiHandler) -> Union[bool, None]:
+        """Ensure client has enough privileges to perform a given request.
+
+        Write a response back to the client if any issue is observed, and the HTTP status may be:
+            * ``401``: if ``Authorization`` header is missing or contain an invalid password;
+            * ``403``: if:
+                * ``restapi.allowlist`` was configured, but client IP is not in the allowed list; or
+                * ``restapi.allowlist_include_members`` is enabled, but client IP is not in the members list; or
+                * a client certificate is expected by the server, but is missing in the request.
+
+        :param rh: the request which access should be checked.
+
+        :returns: ``True`` if client access verification succeeded, otherwise ``None``.
+        """
         if self.__allowlist or self.__allowlist_include_members:
             incoming_ip = ip_address(rh.client_address[0])
             if not any(incoming_ip in net for net in self.__allowlist + tuple(self.__members_ips())):
                 return rh._write_response(403, 'Access is denied')
 
         if not hasattr(rh.request, 'getpeercert') or not rh.request.getpeercert():  # valid client cert isn't present
+            # pyright -- ``__ssl_options`` is initially created as ``None``, but right after that it is replaced with a
+            # dictionary through :func:`reload_config`.
+            assert isinstance(self.__ssl_options, dict)
```

Comment:
If we declare self.__ssl_options as `self.__ssl_options: Dict[str, Any] = {}` there will be no need to do asserts.

Reply:
I wonder if this is true of all these. Having the `assert` statements like this is strange to me, afaik `assert` should only be used in tests.